### PR TITLE
Allow full color tinting on grayscale template images.

### PR DESCRIPTION
**Pull request by jessietea**
As an optimization, `imageRendererFormat` decides on a color space based on the image. Since template images are meant to be black and transparent, this means they are assigned a grayscale color space. This is not ideal, because a template image should be able to be tinted to any color, not just grayscale. In order to allow this, we can simply use the default `UIGraphicsImageRendererFormat` if the image is a template image.

### CODE REVIEW

File: Source/ASImageNode.mm [hunk 1]
```diff
@@ -594,6 +594,14 @@ - (void)setNeedsDisplayWithCompletion:(void (^ _Nullable)(BOOL canceled))display
   [self setNeedsDisplay];
 }
 
+- (void)tintColorDidChange
+{
+  [super tintColorDidChange];
+  if (self.image.renderingMode == UIImageRenderingModeAlwaysTemplate) {
```
Comment (rahul-malik) [inline Source/ASImageNode.mm]: use `_image` here instead of `self.image`

File: Source/Details/ASGraphicsContext.mm [hunk 2]
```diff
@@ -47,7 +47,15 @@ NS_INLINE void ASConfigureExtendedRange(UIGraphicsImageRendererFormat *format)
 
       UIGraphicsImageRendererFormat *format;
       if (sourceImage) {
-        format = sourceImage.imageRendererFormat;
+        if (sourceImage.renderingMode == UIImageRenderingModeAlwaysTemplate) {
+          // Template images will be black and transparent, so if we use
```
Comment (rahul-malik) [inline Source/Details/ASGraphicsContext.mm]: Nice comment!

**Comment by rahul-malik** [comment]
@jessietea - Can we add a grayscale template image in the test resources directory? https://github.com/TextureGroup/Texture/tree/master/Tests/TestResources

Then we can have a new snapshot test that tints that asset with the `ASExperimentalDrawingGlobal` feature enabled

**Comment by rahul-malik** [comment]
@nguyenhuy - it seems like this is stuck on waiting for Danger to report back. is there any way to get past this?

**Comment by nguyenhuy** [comment]
@rahul-malik Danger stopped working because we removed the BuildKite integration. I’m trying to fix it by running Danger with GitHub Actions (#1635).

**Comment by nguyenhuy** [comment]
@rahul-malik Danger is back up now.
@jessietea Can you please rebase with master to pick up the new GitHub Actions workflow? Thanks.

**Pull request merged**